ledger/artifactmanager/blockhistory.go

```diff
@@ -0,0 +1,109 @@
+package artifactmanager
+
+import (
+	"context"
+	"errors"
+	"github.com/insolar/insolar/core"
+	"github.com/insolar/insolar/core/message"
+	"github.com/insolar/insolar/core/reply"
+	"github.com/insolar/insolar/ledger/record"
+)
+
+// HistoryIterator is used to iterate object history.
+//
+// During iteration object refs will be fetched from remote source.
+type HistoryIterator struct {
+	ctx        context.Context
+	messageBus core.MessageBus
+	object     core.RecordRef
+	chunkSize  int
+	fromPulse  *core.PulseNumber
+	fromChild  *core.RecordID
+	buff       []record.ObjectState
```

Comment:
It's best not to store internal structs here. Let's make a separate data struct. It will likely be exported in core.